packages/notifier/src/storesub.js

```diff
@@ -2,9 +2,92 @@
 import { E } from '@endo/eventual-send';
 import { Far, makeMarshal } from '@endo/marshal';
 import { observeIteration } from './asyncIterableAdaptor.js';
+import { makePublishKit, subscribeEach } from './publish-kit.js';
 import { makeSubscriptionKit } from './subscriber.js';
 
+/**
+ * @template T
+ * @param {Subscriber<T>} subscriber
+ * @param {(v: T) => void} consumeValue
+ */
+const forEachPublicationRecord = async (subscriber, consumeValue) => {
+  const iteratorP = E(subscribeEach(subscriber))[Symbol.asyncIterator]();
+
+  let finished = false;
+  while (!finished) {
+    // Allow nested awaits (in loop) because it's safe for each to run in a turn
+    // eslint-disable-next-line no-await-in-loop, @jessie.js/no-nested-await
+    const { value, done } = await E(iteratorP).next();
+    consumeValue(value);
+    finished = !!done;
+  }
+};
+
 /**
+ * Begin iterating the source, storing serialized iteration values.  If the
+ * storageNode's `setValue` operation rejects, the iteration will be terminated.
```

Comment:
```suggestion
 * Begin iterating the source, storing serialized iteration values.  If the
 * storageNode's `setValue` operation rejects, no further writes to it will
 * be attempted (but results will remain available from the subscriber).
```